### PR TITLE
Add net.damonlynch.rapid_photo_downloader

### PR DESCRIPTION
**Pull request by yodatak**
### Please confirm your submission meets all the criteria

- [ X] Please describe the application briefly.
      Rapid Photo Downloader imports photos and videos from cameras, phones,
      memory cards and other devices at high speed. It can be configured to
      rename photos and videos with meaningful filenames you specify. It can also
      back up photos and videos as they are downloaded. It downloads from and backs
      up to multiple devices simultaneously.

- [X] The domain used for the application ID is [controlled by the application developers][appid-domain] and the [application id guidelines][appid] are followed.
- [X ] I have read the and followed all the [Submission and licence requirements][reqs].
- [ X] I have [built][build] and tested the submission locally.
- [ X] I am an author/developer/upstream contributor of the project. If not, I contacted upstream developers about this submission. **Link:**
- https://github.com/damonlynch/rapid-photo-downloader/pull/111

[appid-domain]: https://docs.flathub.org/docs/for-app-authors/requirements/#control-over-domain-or-repository
[appid]: https://docs.flathub.org/docs/for-app-authors/requirements#application-id
[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[build]: https://docs.flathub.org/docs/for-app-authors/submission/#before-submission

### CODE REVIEW

File: exiftool.yaml [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+        url: https://github.com/exiftool/exiftool.git
+        tag: '12.96'
+        commit: 0f63a780906abcccba796761fc2e66a0737e2f16
+        x-checker-data:
+          type: git
+          tag-pattern: ^(\d+\.\d+[02468])$
+    modules:
+      - name: perl
+        buildsystem: simple
+        build-commands:
```
Comment (bbhtt) [inline exiftool.yaml:40]: This should use the autotools buildsystem

File: gudev.json
Comment (bbhtt) [inline gudev.json:1]: Use this from shared modules https://github.com/flathub/shared-modules?tab=readme-ov-file#adding

File: libmediainfo.yaml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: libmediainfo
+build-commands: []
+buildsystem: simple
+modules:
+  - name: libzen
+    buildsystem: simple
+    build-commands:
+      - cd Project/CMake && cmake -DCMAKE_INSTALL_PREFIX=/app -DCMAKE_INSTALL_LIBDIR=lib -DCMAKE_BUILD_TYPE=Release
+      - cd Project/CMake && make
+      - cd Project/CMake && make install
```
Comment (bbhtt) [inline libmediainfo.yaml:10]: why does this use make and cmake both ?

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+  - python-arrow.yaml
+  # mega runtime dep
+  - python-runtimedep.yaml
+
+  # for camaera capacity
+  - python-gphoto2.yaml
+
+  # for camaera capacity
+  - python-pyzmq.yaml
+
```
Comment (hfiguiere) [inline net.damonlynch.rapid_photo_downloader.yaml:101]: manifest that are not auto generated should NOT be separate and be inlined here. Otherwise it is unmaintainable.

File: libmediainfo.yaml (hunk 2 above)
Comment (bbhtt) [reply libmediainfo.yaml:10]: and same for the one below

File: net.damonlynch.rapid_photo_downloader.desktop
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.desktop:1]: Desktop file should be upstream

File: net.damonlynch.rapid_photo_downloader.metainfo.xml
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.metainfo.xml:1]: Metainfo should be upstream

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+  - /lib/pkgconfig
+  - /man
+  - /share/doc
+  - /share/gtk-doc
+  - /share/man
+  - /share/vala
+  - "*.a"
+  - "*.la"
+  - "/lib/*.a"
+  - "/lib/*.la"
```
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.yaml:20]:
```suggestion
```

the two above does the same thing

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+  - "/lib/pkgconfig"
+  - "/include"
+  - "/lib/cmake"
```
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.yaml:23]:
```suggestion
```

dupes

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+  - --filesystem=/media
+  - --filesystem=/run/media
+  - --filesystem=/mnt
+
+  # notifications
+  - --talk-name=org.freedesktop.Notifications
+  - --talk-name=org.kde.StatusNotifierWatcher
+
+  # file chooser
+  - --talk-name=org.freedesktop.portal.FileChooser
```
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.yaml:51]:
```suggestion
```

You don't need to specify this, it works automatically

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 7]
```diff
@@ -0,0 +1,153 @@
...
+
+  # App indicator ( not sure if needed)
+  - --talk-name=com.canonical.AppMenu.Registrar
+  - --talk-name=com.canonical.Unity
+  - --talk-name=com.canonical.indicator.application
+  - --talk-name=org.ayatana.indicator.application
```
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.yaml:57]: Can you make sure ? There isn't a point in adding permissions if they aren't required, it opens up more sandbox holes.

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 8]
```diff
@@ -0,0 +1,153 @@
...
+
+  # Not sure if needed ?
+  # - --unset-env=XDG_DATA_HOME
```
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.yaml]:
```suggestion
```

Never do this

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 9]
```diff
@@ -0,0 +1,153 @@
...
+
+  # QT and perl libs
+  - --env=QT_PLUGIN_PATH=/app/lib64/plugins:/app/lib/plugins
```
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.yaml:60]: `/app/lib64` should not exist. You should install to `/app/lib` always.

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 10]
```diff
@@ -0,0 +1,153 @@
...
+  - --env=QTWEBENGINEPROCESS_PATH=/app/bin/QtWebEngineProcess
+  - --env=PERL5LIB=/app/lib/perl5/site_perl/5.36.0/
+
+  # access to udisk
+  - --system-talk-name=org.freedesktop.UDisks2
+
+  # needed for introspection ?
+  - --talk-name=org.freedesktop.DBus.Introspectable.*
```
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.yaml:68]:
```suggestion
```

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 11]
```diff
@@ -0,0 +1,153 @@
...
+  # runtime dep
+  - python-arrow.yaml
+  # mega runtime dep
+  - python-runtimedep.yaml
+
+  # for camaera capacity
+  - python-gphoto2.yaml
+
+  # for camaera capacity
+  - python-pyzmq.yaml
```
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.yaml:97]: except python dependencies that are autogenerated with flatpak-pip-generator rest should be in the main manifest.

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 12]
```diff
@@ -0,0 +1,153 @@
...
+  - --filesystem=xdg-pictures
+  - --filesystem=xdg-videos
+
+  # external disk mount
+  - --filesystem=/media
+  - --filesystem=/run/media
+  - --filesystem=/mnt
+
+  # notifications
+  - --talk-name=org.freedesktop.Notifications
```
Comment (bbhtt) [inline net.damonlynch.rapid_photo_downloader.yaml:47]:
```suggestion
```

libnotify uses the portal automatically

File: net.damonlynch.rapid_photo_downloader.yaml [hunk 13]
```diff
@@ -0,0 +1,153 @@
...
+
+  # App indicator ( not sure if needed)
+  - --talk-name=com.canonical.AppMenu.Registrar
+  - --talk-name=com.canonical.Unity
+  - --talk-name=com.canonical.indicator.application
+  - --talk-name=org.ayatana.indicator.application
+
+  # Not sure if needed ?
+  # - --unset-env=XDG_DATA_HOME
+
```
Comment (hfiguiere) [inline net.damonlynch.rapid_photo_downloader.yaml:61]:
```suggestion
```

Setting this is an error.

File: net.damonlynch.rapid_photo_downloader.desktop [hunk 14]
```diff
@@ -0,0 +1,14 @@
+[Desktop Entry]
+Categories=Graphics;Photography;
+Comment=Download, rename and back up photos and videos from cameras and other devices
+Exec=rapid-photo-downloader %f
+GenericName=Photo Downloader
+Icon=net.damonlynch.rapid_photo_downloader.svg
```
Comment (hfiguiere) [inline net.damonlynch.rapid_photo_downloader.desktop:6]: this is icon *name* not file name.
```suggestion
Icon=net.damonlynch.rapid_photo_downloader
```

this file should be upstream anyway

File: libmediainfo.yaml [hunk 15]
```diff
@@ -0,0 +1,36 @@
+name: libmediainfo
+build-commands: []
+buildsystem: simple
+modules:
+  - name: libzen
+    buildsystem: simple
```
Comment (hfiguiere) [inline libmediainfo.yaml:6]: this should me `buildsystem: cmake-ninja`

File: gudev.json
Comment (yodatak) [reply gudev.json:1]: I was using it but i need     "-Dintrospection=enabled", for make it works with the app ( maybe it didnt understand well the app)

File: net.damonlynch.rapid_photo_downloader.yaml (hunk 11 above)
Comment (yodatak) [reply net.damonlynch.rapid_photo_downloader.yaml:97]: They are not autogenrated because the projet  use hatch and the thereis no longer file to use with pip to generate the dep https://github.com/flatpak/flatpak-builder-tools/issues/419

File: gudev.json
Comment (hfiguiere) [inline gudev.json:1]: this should be inline in the manifest.

File: exiftool.yaml [hunk 16]
```diff
@@ -0,0 +1,87 @@
...
+      - type: archive
+        url: https://github.com/Exiv2/exiv2/archive/refs/tags/v0.28.3.tar.gz
+        sha256: 1315e17d454bf4da3cc0edb857b1d2c143670f3485b537d0f946d9ed31d87b70
+        x-checker-data:
+          type: anitya
+          project-id: 769
+          stable-only: true
+          url-template: https://github.com/Exiv2/exiv2/archive/refs/tags/v$version.tar.gz
+
+  - name: exiftool
```
Comment (hfiguiere) [inline exiftool.yaml:22]: did you test the package? Perl is in the Sdk but NOT the platform.

File: exiftool.yaml
Comment (hfiguiere) [inline exiftool.yaml:1]: this should be inline in the manifest.

File: exiftool.yaml (hunk 1 above)
Comment (hfiguiere) [reply exiftool.yaml:40]: NO. It's Perl

File: net.damonlynch.rapid_photo_downloader.yaml (hunk 3 above)
Comment (hfiguiere) [reply net.damonlynch.rapid_photo_downloader.yaml:97]: no it doesn't.

File: exiftool.yaml (hunk 16 above)
Comment (yodatak) [reply exiftool.yaml:22]: Yes and without exiftool rapid photo downloader don't work

Comment (hfiguiere) [reply exiftool.yaml:22]: So how does it work?

```
$ flatpak run org.kde.Platform//5.15-23.08
[📦 org.kde.Platform ~]$ perl
sh: perl: command not found
```

tl;dr for Hugin I added Perl to the manifest to build it. Same for other packages that require Perl at runtime.

Comment (hfiguiere) [reply exiftool.yaml:22]: never mind I missed it below.

File: net.damonlynch.rapid_photo_downloader.yaml (hunk 7 above)
Comment (yodatak) [reply net.damonlynch.rapid_photo_downloader.yaml:56]: I remove them it seem not needed

File: exiftool.yaml (hunk 16 above)
Comment (yodatak) [reply exiftool.yaml]: If you find a way to impove it without needed the perl build it could be cool but i dont know where to look

Comment (hfiguiere) [reply exiftool.yaml]: it's all good. I just had missed perl that I was expecting to see before no inside.

File: net.damonlynch.rapid_photo_downloader.yaml (hunk 12 above)
Comment (yodatak) [reply net.damonlynch.rapid_photo_downloader.yaml]: Thanks i made the change

File: net.damonlynch.rapid_photo_downloader.yaml (hunk 10 above)
Comment (yodatak) [reply net.damonlynch.rapid_photo_downloader.yaml]: thanks i made the changed

File: libmediainfo.yaml (hunk 2 above)
Comment (yodatak) [reply libmediainfo.yaml]: Thanks i made the change

File: gudev.json
Comment (yodatak) [reply gudev.json:1]: I did thanks !

File: net.damonlynch.rapid_photo_downloader.metainfo.xml
Comment (yodatak) [reply net.damonlynch.rapid_photo_downloader.metainfo.xml:1]: They already exist but they are not valid when i test 
with flatpak run --command=flatpak-builder-lint org.flatpak.Builder appstream

how can i do to make them upstream ? put in a folder flatpak ?
https://github.com/damonlynch/rapid-photo-downloader/blob/main/data/net.damonlynch.rapid_photo_downloader.metainfo.xml.in

File: net.damonlynch.rapid_photo_downloader.desktop
Comment (yodatak) [reply net.damonlynch.rapid_photo_downloader.desktop:1]: They already exist but they are not valid when i test 
with flatpak-builder-lint
because of _name i think ?

how can i do to make them upstream ? put in a folder flatpak ? 
https://github.com/damonlynch/rapid-photo-downloader/blob/main/data/net.damonlynch.rapid_photo_downloader.desktop.in

File: net.damonlynch.rapid_photo_downloader.desktop (hunk 14 above)
Comment (yodatak) [reply net.damonlynch.rapid_photo_downloader.desktop]: I made the change thanks and for upstream i got some issue because the desktop file is not compilent 
https://github.com/damonlynch/rapid-photo-downloader/blob/main/data/net.damonlynch.rapid_photo_downloader.desktop.in

File: net.damonlynch.rapid_photo_downloader.metainfo.xml
Comment (hfiguiere) [reply net.damonlynch.rapid_photo_downloader.metainfo.xml:1]: patch + submission upstream to fix them.

File: net.damonlynch.rapid_photo_downloader.desktop
Comment (hfiguiere) [reply net.damonlynch.rapid_photo_downloader.desktop:1]: often for desktop file, the `desktop-file-edit` command can work.
Also for icons a top level `rename-icon` in the manifest do fix the desktop file (it rename the icons)

File: gudev.json
Comment (bbhtt) [reply gudev.json:1]: Ok then add it in the main manifest.

**Comment by bbhtt** [comment]
bot, build net.damonlynch.rapid_photo_downloader

**Comment by yodatak** [comment]
The polkit patch is now upstream https://github.com/polkit-org/polkit/pull/496

**Comment by yodatak** [comment]
> Build [149654 failed](https://buildbot.flathub.org/#/builders/6/builds/149654)

Maybe some arm problem

]2;flatpak-builder: Installing libfuseFB: Running 'flatpak build --die-with-parent --env=FLATPAK_BUILDER_BUILDDIR=/run/build/libfuse --nofilesystem=host:reset --filesystem=/srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/build/libfuse-1 --bind-mount=/run/build/libfuse=/srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/build/libfuse-1 --build-dir=/run/build/libfuse --bind-mount=/run/ccache=/srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/ccache --unshare=network --env=SOURCE_DATE_EPOCH=1727317569 '--env=CFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer ' '--env=CXXFLAGS=-O2 -pipe -g -Wp,-D_FORTIFY_SOURCE=2 -Wp,-D_GLIBCXX_ASSERTIONS -fexceptions -fstack-protector-strong -grecord-gcc-switches -fasynchronous-unwind-tables -fstack-clash-protection -fno-omit-frame-pointer ' '--env=LDFLAGS=-L/app/lib -Wl,-z,relro,-z,now -Wl,--as-needed ' --env=UDEV_RULES_PATH=/app/etc/udev/rules.d --env=INIT_D_PATH=/app/etc/init.d --env=MOUNT_FUSE_PATH=/app/bin --env=CCACHE_DIR=/run/ccache/disabled --env=PATH=/app/bin:/usr/bin --env=LD_LIBRARY_PATH=/app/lib --env=PKG_CONFIG_PATH=/app/lib/pkgconfig:/app/share/pkgconfig:/usr/lib/pkgconfig:/usr/share/pkgconfig --env=FLATPAK_BUILDER_N_JOBS=80 /srv/buildbot/worker/build-aarch64-6/build/.flatpak-builder/rofiles/rofiles-p7GYCI make -j80 -l160' on host
Making all in include
make[1]: Entering directory '/run/build/libfuse/include'
make  all-am
make[2]: Entering directory '/run/build/libfuse/include'
make[2]: Leaving directory '/run/build/libfuse/include'
make[1]: Leaving directory '/run/build/libfuse/include'
Making all in lib
make[1]: Entering directory '/run/build/libfuse/lib'
  CC       fuse.lo
  CC       fuse_kern_chan.lo
  CC       fuse_loop_mt.lo
  CC       fuse_loop.lo
  CC       fuse_lowlevel.lo
  CC       fuse_opt.lo
  CC       fuse_mt.lo
  CC       fuse_session.lo
  CC       buffer.lo
  CC       cuse_lowlevel.lo
  CC       helper.lo
  CC       fuse_signals.lo
  CC       mount.lo
  CC       mount_util.lo
  CC       ulockmgr.lo
  CC       modules/iconv.lo
  CC       modules/subdir.lo
In file included from fuse_loop_mt.c:11:
../include/fuse_kernel.h:93:15: error: conflicting types for ‘int64_t’; have ‘long long int’
In file included from fuse_loop_mt.c:11:
../include/fuse_kernel.h:93:15: error: conflicting types for ‘int64_t’; have ‘long long int’

**Comment by hadess** [comment]
> Maybe some arm problem

https://github.com/libfuse/libfuse/commit/914871b20a901e3e1e981c92bc42b1c93b7ab81b.patch
should fix it
